docs/enhancements/infrastructure-monitoring-service-api/lifecycle.md

```diff
@@ -0,0 +1,530 @@
+|-------------------------------------------------|-----------------|-----------------------------------------------|--------------------------------------------------------------------------|-------------------|
+| `internal/v1/caas-alerts/alertmanager`          | POST            | Alertmanager notifications comes through here | https://prometheus.io/docs/alerting/latest/configuration/#webhook_config | None              |
+| `internal/v1/hardware-alerts/{hw-vendor-name}}` | POST            | TBD                                           | TBD                                                                      | TBD               |
+
+
+
+### `alarms` family
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarms` with GET
+1. Get active alarms from `alarm_event_record` table
+3. Response with retrieved list of AlarmEventRecord
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarms/{alarmEventRecordId}` with GET
+1. Client calls with an AlarmEventRecordID
+2. Get active alarms from DB from `alarm_event_record`
+3. Response with retrieved instance of AlarmEventRecord
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarms/{alarmEventRecordId}` with PATCH
+1. Client calls with an AlarmEventRecordID and `AlarmEventRecordModifications` as patch payload
+2. If AlarmEventRecordModifications.alarmAcknowledged is True, update `alarm_event_record` table
+3. Response with `AlarmEventRecordModifications`
+
+### `alarmSubscriptions` family
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions` with GET
+1. Query the storage `alarm_subscription_info`
+3. Response with a list of AlarmSubscriptions
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions` with POST
+1. Client calls with `AlarmSubscriptionInfo` payload
+2. Query the storage `alarm_subscription_info` 
+3. Response with AlarmSubscriptionInfo
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions/{alarmSubscriptionId}` with GET
+1. Client calls with an `alarmSubscriptionId`
+2. Query the storage `alarm_subscription_info`
+3. Response with retrieved instance of AlarmSubscriptionInfo
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions/{alarmSubscriptionId}` with GET
+1. Client calls with an `alarmSubscriptionId`
+2. Query the storage `alarm_subscription_info` with delete
+3. No special response (just error code if applicable)
+
+### `internal` family
+
+The APIs are not exposed to client and available only internally 
+
+### Steps `internal/v1/caas-alerts/alertmanager`
+This API is activated only when after configuring ACM alertmanager to call back alarm service
+
+A minimal configuration
+```yaml
+route:
+  receiver: webhook_receiver
+receivers:
+  - name: webhook_receiver
+    webhook_configs:
+      - url: "o-ran-inventory-api-alarms.kubernetes.svc/internal/v1/caasAlerts/alertmanager"
+        send_resolved: true 
```

Comment:
We need to discuss with Brent about if we should filter out the OCP management alert we talked earlier today.

Reply:
like we spoke about it internally...not doing this